### PR TITLE
autodetect config type

### PR DESCRIPTION
**Pull request by emcek**
looks like `config_type` parameter in `load_config()` is redundant. I've add `*args` and `**kwargs` for backward compatibility.

### CODE REVIEW

File: moler/config/__init__.py [hunk 1]
```diff
@@ -64,13 +64,15 @@ def configs_are_same(config_list, config_to_find):
...
-    Load Moler's configuration from config file
+    Load Moler's configuration from config file.
 
-    :param config: either dict or config filename directly provided (overwrites 'from_env_var' if both given)
+    config_type parameter is deprached and since 1.4.0 not need to provide. Type of config is autodetecting.
+
+    :param config: either dict or config filename directly provided (overwrites 'from_env_var' if both given).
+    :type config: dict or str
     :param from_env_var: name of environment variable storing config filename
-    :param config_type: 'dict' ('config' param is dict) or 'yaml' ('config' is filename of file with YAML content)
```
Comment (greg-latuszek) [inline moler/config/__init__.py]: You are removing `config_type` parameter so you should not mention it in documentation - that confuses users.
Deprecation should be put into `CHANGELOG.md` as:
```
### Deprecated
* `config_type` parameter of `load_config()` is not needed, configuration type is autodetected
```
So, as docstring I would put:
```
 """
    Load Moler's configuration from config file.
    :param config: either dict or config filename directly provided (overwrites 'from_env_var' if both given).
    :type config: dict or str
    :param from_env_var: name of environment variable storing config filename (file is in YAML format)
    :return: None
"""
```

File: moler/config/__init__.py [hunk 2]
```diff
@@ -85,21 +87,18 @@ def load_config(config=None, from_env_var=None, config_type='yaml'):
         add_devices_only = True
         loaded_config.append(config)
 
-    if (config_type != 'dict') and (config_type != 'yaml'):  # no other format supported yet
-        raise WrongUsage("Unsupported config_type: '{}'. Allowed are: 'dict' or 'yaml'.".format(config_type))
+    if config is None and from_env_var is None:
+        raise WrongUsage("Provide either 'config' or 'from_env_var' parameter (none given).")
+    elif not from_env_var and (not isinstance(config, six.string_types) and not isinstance(config, dict)):  # no other format supported yet
+        raise WrongUsage("Unsupported config type: '{}'. Allowed are: 'dict' or 'yaml'.".format(type(config)))
```
Comment (greg-latuszek) [inline moler/config/__init__.py]: `Allowed are: 'dict' or 'yaml'` suggests that you can provide config as yaml string. Confusing.
change to:
`Allowed are: 'dict' or 'str' holding config filename (file is in YAML format)`

File: moler/config/__init__.py [hunk 3]
```diff
@@ -85,21 +87,18 @@ def load_config(config=None, from_env_var=None, config_type='yaml'):
         add_devices_only = True
         loaded_config.append(config)
 
-    if (config_type != 'dict') and (config_type != 'yaml'):  # no other format supported yet
-        raise WrongUsage("Unsupported config_type: '{}'. Allowed are: 'dict' or 'yaml'.".format(config_type))
+    if config is None and from_env_var is None:
+        raise WrongUsage("Provide either 'config' or 'from_env_var' parameter (none given).")
+    elif not from_env_var and (not isinstance(config, six.string_types) and not isinstance(config, dict)):  # no other format supported yet
```
Comment (greg-latuszek) [inline moler/config/__init__.py]: wrong check - what if `from_env_var="AAA"` and `config=[1,2,3]`? It won't be detected.
condition should be:
`elif config and (not isinstance(config, six.string_types) and not isinstance(config, dict)):`

File: test/device/test_device_configuration.py [hunk 4]
```diff
@@ -446,11 +446,13 @@ def test_cannot_load_config_from_when_path_or_from_env_var_not_provide(moler_con
     assert "Provide either 'config' or 'from_env_var' parameter (none given)" in str(err.value)
 
 
-def test_cannot_load_config_from_when_wrong_config_type_provided(moler_config):
+@pytest.mark.parametrize('params', [{'config': (), 'config_type': "wrong_type"},
+                                    {'config': ()}])
+def test_cannot_load_config_from_when_wrong_config_type_provided(moler_config, params):
```
Comment (greg-latuszek) [inline test/device/test_device_configuration.py]: add comment: `{'config': (), 'config_type': "wrong_type"},  # test backward compatibility`

File: test/device/test_device_configuration.py [hunk 5]
```diff
@@ -446,11 +446,13 @@ def test_cannot_load_config_from_when_path_or_from_env_var_not_provide(moler_con
...
     with pytest.raises(WrongUsage) as err:
-        moler_config.load_config(config_type="wrong_type")
+        moler_config.load_config(**params)
 
-    assert "Unsupported config_type" in str(err.value)
+    assert "Unsupported config type" in str(err.value)
```
Comment (greg-latuszek) [inline test/device/test_device_configuration.py]: missing test for case where both params provided and config has wrong type:
` from_env_var="AAA" and config=[1,2,3]`

File: moler/config/__init__.py (hunk 1 above)
Comment (Ernold11) [reply moler/config/__init__.py]: **kwargs is not enought. What if someone use load_config and pass to it arguments positional without parameter_name?

File: test/device/test_device_configuration.py (hunk 4 above)
Comment (emcek) [reply test/device/test_device_configuration.py]: 426500832b29ff17d97c4a5570371071682192af

File: test/device/test_device_configuration.py (hunk 5 above)
Comment (emcek) [reply test/device/test_device_configuration.py]: 0fef3d93c7b7e81dfec3d4ae2e826dbe5d57d139

File: moler/config/__init__.py (hunk 3 above)
Comment (emcek) [reply moler/config/__init__.py]: 0fef3d93c7b7e81dfec3d4ae2e826dbe5d57d139

File: moler/config/__init__.py (hunk 2 above)
Comment (emcek) [reply moler/config/__init__.py]: 5f07205bb653e034a28e4baa13c691d023b1f200

File: moler/config/__init__.py (hunk 1 above)
Comment (emcek) [reply moler/config/__init__.py]: 15d257514b4964fd6bb7612543ec8460f3b02295 and 8d14ba8e68c42cdbae7e0dae8e0d7113e3fad0ab

**Comment by emcek** [comment]
Ready for final review

**Pull request merged**